src/NuGet.Clients/NuGet.CommandLine/Commands/Pack/AssemblyMetadataExtractor.cs

```diff
@@ -86,7 +88,14 @@ public AssemblyMetadata GetMetadata(string path)
                     string assemblyInformationalVersion = GetAttributeValueOrDefault<AssemblyInformationalVersionAttribute>(attributes);
                     if (!NuGetVersion.TryParse(assemblyInformationalVersion, out version))
                     {
-                        version = NuGetVersion.Parse(assemblyName.Version.ToString());
+                        if (string.IsNullOrEmpty(assemblyInformationalVersion))
+                        {
+                            version = NuGetVersion.Parse(assemblyName.Version.ToString());
+                        }
+                        else
+                        {
+                            throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, NuGetResources.Error_AssemblyInformationalVersion, assemblyInformationalVersion));
```

Comment:
Usually ArgumentException is for function arguments. Consider changing this, but it doesn't matter too much.